src/dune_cache/local.ml

```diff
@@ -5,8 +5,8 @@ open Import
 
 module Store_artifacts_result = struct
   type t =
-    | Stored of (Path.Build.t * Digest.t) list
-    | Already_present of (Path.Build.t * Digest.t) list
+    | Stored of (Path.Build.t * Digest.t * Path.Build.t option) list
+    | Already_present of (Path.Build.t * Digest.t * Path.Build.t option) list
```

Comment:
It's hard to understand what this argument is. I suggest turning this tuple into a record.

Reply:
I turned that into an `Artifact` module which seems useful.